### DISCUSSION TITLE
Automation state per track on OLED

### DISCUSSION THREAD

**Discussion by Barrylimestone** [tttapa/Control-Surface#442] [Q&A]
Hi, 

I want to make a scribble strip (with or without a button for automation control options, I have a built in hui automation system) for my console. I already discovered Henk Meid's project which is really cool!

Hypothetically speaking, could I connect 60 OLEDS and display track names on them using the library and MCU protocol?

(I will have two different DAWS, one for automation, one for tracks etc.)

I Was also wondering (seeing VU meter, panning, and other things can be displayed) whether it would be possible to show automation mode per channel on the OLED displays, perhaps someone has tried this before?

(This is the idea, I photoshopped it, if it is possible to display automation info on the oleds alongside track names, I won't need the leds etc.)

https://www.dropbox.com/s/7r3q8wb1cxpfcsz/Idea%20SCribble%20Strip3-min.jpg?dl=0

Thanks for reading!

I appreciate the suggestions, being a noob (I know making this is very difficult).

Cheers,

Folly

**Comment by tttapa** [comment]
> Hypothetically speaking, could I connect 60 OLEDS and display track names on them using the library and MCU protocol?

There are two main limitations: how many displays you can drive using the display library you're using and how long that takes, and how many tracks you can monitor using the MCU protocol.

AFAIK, the Adafruit_SSD1306 library does not allow you to share the display buffers (I do have a for that supports it, it's pretty easy to modify). 60 displays will take a considerable amount of time to draw.

The MCU protocol only allows you to use 8 tracks at any given time. This number can be increased by using extenders and banks, but it's cumbersome. It would be much easier to use a different (custom?) protocol if your DAW supports it.

> whether it would be possible to show automation mode per channel on the OLED displays 

If your DAW sends this information, Control Surface can display it. I don't think the MCU protocol supports this though, so you'll have to use a different protocol.

So the main thing to consider is the protocol used by your DAW(s), which is not something I can help you with right now. If you know what kind of data your DAW will be sending, I can tell you how to receive and display it using the Control Surface library.
